### DISCUSSION TITLE
Localization of Blog and Blog-Posts, when only some posts have localized versions...

### DISCUSSION THREAD

**Discussion by bashuss** [OrchardCMS/OrchardCore#16516] [Q&A] [answered]
Hi all,

I would like my blog page to show all blog posts, but dependent on the language of the blog page, I want to show the localized version of a post, if available. If not I still want to show the post, but in the original language.

Is this possible purely with localization features, do I have to use queries to make that possible, or am I missing a third way?

You are also invited to tell me, if you think, this is a bad idea. I have still not completely decided, if it would not simply be enough to say: for more posts in other languages, use the the culture picker in the menu bar... 

My thought was, that there won't be so many posts - especially in some languages. It will look very sad, if there is nearly nothing to be found.

**Comment by hishamco** [comment, marked as answer]
Currently if you enabled content localization module and add localize part to the post you will be able to see it with respect of the selected culture

If you want to see all the versions or non localized version you might have extra work. FYI I demo a localization matrix long time ago, I am sure there's some related work in Orchard Core Contrib (OCC)

**Reply by bashuss** [reply]
Thanks, I worked out an SqlQuery, that does the job. 
```
SELECT 
 ABS(MAX(bp.DocumentId * IsCurrentCulture)) 
 AS DocumentId
FROM
(Select 
 lci.DocumentId,
 IIF(lci.Culture=@culture:"de-de", 1, -1) AS IsCurrentCulture, 
 LocalizationSet 
 From LocalizedContentItemIndex lci
 Inner Join ContentItemIndex ci
 On lci.DocumentId = ci.DocumentId
 where 
  ci.Published = true 
  and ci.ContentType = 'BlogPost') AS bp
GROUP BY bp.LocalizationSet
```
It may work in a more elegant way though. I could not find a way to do it with lucene queries, but I simple do not understand the concept completely.

Unfortunately there seems to be an issue with calling the sql query from a liquid template - maybe some interaction with lucene queries module: Only lucene queries are found, when lucene feature is active. With lucene feature turned off, the liquid code and the backend creates error for queries menu entry. I'll move that to an issue.

**Reply by bashuss** [reply]
https://github.com/OrchardCMS/OrchardCore/issues/16525
